rir/src/compiler/transform/bb.cpp

```diff
@@ -195,22 +196,35 @@ BB* BBTransform::lowerExpect(Code* code, BB* src, BB::Instrs::iterator position,
     return split;
 }
 
-void BBTransform::insertAssume(Value* condition, Checkpoint* cp, BB* bb,
-                               BB::Instrs::iterator& position,
-                               bool assumePositive) {
-    position = bb->insert(position, (Instruction*)condition);
-    auto assume = new Assume(condition, cp);
-    if (!assumePositive)
-        assume->Not();
-    position = bb->insert(position + 1, assume);
-    position++;
-};
-
-void BBTransform::insertAssume(Value* condition, Checkpoint* cp,
-                               bool assumePositive) {
+Assume* BBTransform::insertAssume(Instruction* condition, Checkpoint* cp,
+                                  BB* bb, BB::Instrs::iterator& position,
+                                  bool assumePositive) {
+    if (Parameter::RIR_UNSOUND_ASSUME) {
+        // don't actually insert the assume
+        delete condition; // we don't actually need it
+        return NULL;
```

Comment:
but this might silently produce a wrong result, which is not the same as "crashing"